js/ui/validationMessage.js

```diff
@@ -0,0 +1,134 @@
+            integrationOptions: {},
+            templatesRenderAsynchronously: false,
+            shading: false,
+            width: 'auto',
+            height: 'auto',
+            closeOnOutsideClick: false,
+            closeOnTargetScroll: false,
+            animation: null,
+            visible: true,
+            propagateOutsideClick: true,
+            _checkParentVisibility: false,
+            rtlEnabled: false,
+            contentTemplate: this._renderInnerHtml,
+            maxWidth: '100%',
+
+            mode: 'auto',
+            validationErrors: undefined,
+            positionRequest: undefined,
+            boundary: undefined,
+            offset: { h: 0, v: 0 }
+        });
+    },
+
+    _init() {
+        this.callBase();
+        this.updateMaxWidth();
+        this._updatePosition();
+    },
+
+    _initMarkup() {
+        this.callBase();
+
+        this.$element().addClass(INVALID_MESSAGE);
+        this._wrapper().addClass(INVALID_MESSAGE);
+        this._toggleModeClass();
+
+        this.$content()
+            .addClass(INVALID_MESSAGE_CONTENT)
+            .attr('id', 'dx-' + new Guid());
+    },
+
+    _renderInnerHtml(element) {
+        const $element = element && $(element);
+        const validationErrors = this.option('validationErrors') || [];
+        let validationErrorMessage = '';
+        validationErrors.forEach((err) => {
+            const separator = validationErrorMessage ? '<br />' : '';
+            validationErrorMessage += separator + encodeHtml(err?.message || '');
+        });
+
+        $element?.html(validationErrorMessage);
+    },
+
+    _toggleModeClass() {
+        const mode = this.option('mode');
+        this.$element()
+            .toggleClass(INVALID_MESSAGE_AUTO, mode === 'auto')
+            .toggleClass(INVALID_MESSAGE_ALWAYS, mode === 'always');
+    },
+
+    updateMaxWidth() {
+        const targetWidth = this.option('target')?.outerWidth();
+        let maxWidth = '100%';
+        if(targetWidth) {
+            maxWidth = Math.max(targetWidth, VALIDATION_MESSAGE_MIN_WIDTH);
+        }
+
+        this.option('maxWidth', maxWidth);
+    },
+
+    _updatePosition: function() {
+        const positionRequest = this.option('positionRequest');
+        const rtlEnabled = this.option('rtlEnabled');
+        const positionSide = getDefaultAlignment(rtlEnabled);
+        const offset = this.option('offset');
+        const verticalPositions = positionRequest === 'below' ? [' top', ' bottom'] : [' bottom', ' top'];
+
+        if(rtlEnabled) offset.h = -offset.h;
+        if(positionRequest !== 'below') offset.v = -offset.v;
+
+        this.option('position', {
+            offset,
+            boundary: this.option('boundary'),
+            my: positionSide + verticalPositions[0],
+            at: positionSide + verticalPositions[1],
+            collision: 'none flip'
+        });
+    },
+
+    _optionChanged(args) {
+        switch(args.name) {
```

Comment:
```suggestion
        const { name, value } = args;
        switch(name) {
```

and replace `args.value` with `value` below

Reply:
fixed